flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1384,4 +1389,27 @@ public void run() {
 			});
 		}
 	}
+
+	private void failPendingCheckpoint(
+			final PendingCheckpoint pendingCheckpoint,
+			final CheckpointFailureReason reason,
+			final Throwable cause) {
+
+		CheckpointException exception = new CheckpointException(reason, cause);
+
+		if (cause != null) {
+			pendingCheckpoint.abort(reason, cause);
```

Comment:
When I look into the implementation of the two abort methods that are distinguished here, the `if` block seems unnecessary, I would just always call `pendingCheckpoint.abort(reason, cause);` and probably annotate the `cause` parameter as `@Nullable`